src/hardware/mouse/mouse_manymouse.cpp

```diff
@@ -0,0 +1,481 @@
+{
+	if (GCC_UNLIKELY(event.device >= mouse_info.physical.size()))
+		return; // device ID out of supported range
+	if (GCC_UNLIKELY(mouse_config.no_mouse &&
+	                 event.type != MANYMOUSE_EVENT_DISCONNECT))
+		return; // mouse control disabled in GUI
+
+	const auto device_idx = static_cast<uint8_t>(event.device);
+	const auto interface_id = physical_devices[device_idx].GetMappedInterfaceId();
+	const bool no_interface = (interface_id == MouseInterfaceId::None);
+
+	switch (event.type) {
+	case MANYMOUSE_EVENT_ABSMOTION:
+		// LOG_INFO("MANYMOUSE #%u ABSMOTION axis %d, %d", event.device,
+		// event.item, event.value);
+		break;
+
+	case MANYMOUSE_EVENT_RELMOTION:
+		// LOG_INFO("MANYMOUSE #%u RELMOTION axis %d, %d", event.device,
+		// event.item, event.value);
+		if (no_interface || critical_only)
+			break; // movements not relevant at this moment
+		if (event.item != 0 && event.item != 1)
+			break; // only movements related to x and y axis are
+			       // relevant
+
+		if (rel_x.size() <= device_idx) {
+			rel_x.resize(static_cast<size_t>(device_idx + 1), 0);
+			rel_y.resize(static_cast<size_t>(device_idx + 1), 0);
+		}
+
+		if (event.item)
+			rel_y[event.device] += event.value; // event.item 1
+		else
+			rel_x[event.device] += event.value; // event.item 0
+		break;
+
+	case MANYMOUSE_EVENT_BUTTON:
+		// LOG_INFO("MANYMOUSE #%u BUTTON %u %s", event.device,
+		// event.item, event.value ? "press" : "release");
+		if (no_interface || (critical_only && !event.value) ||
+		    (event.item >= max_buttons))
+			// TODO: Consider supporting extra mouse buttons
+			// in the future. On Linux event items 3-7 are for
+			// scroll wheel(s), 8 is for SDL button X1, 9 is
+			// for X2, etc. - but I don't know yet if this
+			// is consistent across various platforms
+			break;
+		MOUSE_EventButton(static_cast<uint8_t>(event.item),
+		                  event.value,
+		                  interface_id);
+		break;
+
+	case MANYMOUSE_EVENT_SCROLL:
+		// LOG_INFO("MANYMOUSE #%u WHEEL #%u %d", event.device,
+		// event.item, event.value);
+		if (no_interface || critical_only || (event.item != 0))
+			break; // only the 1st wheel is supported
+		MOUSE_EventWheel(clamp_to_int16(-event.value), interface_id);
+		break;
+
+	case MANYMOUSE_EVENT_DISCONNECT:
+		// LOG_INFO("MANYMOUSE #%u DISCONNECT", event.device);
+		physical_devices[event.device].disconnected = true;
+		for (uint8_t button = 0; button < max_buttons; button++)
+			MOUSE_EventButton(button, false, interface_id);
+		MOUSE_NotifyDisconnect(interface_id);
+		break;
+
+	default:
+		// LOG_INFO("MANYMOUSE #%u (other event)", event.device);
+		break;
+	}
+}
+
+void ManyMouseGlue::Tick()
+{
+	assert(!mouse_config.no_mouse);
+
+	// Handle all the events from the queue
+	ManyMouseEvent event;
+	while (ManyMouse_PollEvent(&event))
+		HandleEvent(event);
+
+	// Report accumulated mouse movements
+	for (uint8_t idx = 0; idx < rel_x.size(); idx++) {
```

Comment:
Suggest using a `size_t` for idx, or asserting that `rel_x.size()` is no bigger than `UINT8_MAX`.

Reply:
Added assertion.